cvat-ui/src/utils/preview-queue.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { Dispatch } from 'redux';
+import {
+    Job, Task, Project, CloudStorage,
+} from 'cvat-core-wrapper';
+import MLModel from 'cvat-core/src/ml-model';
+import { getJobPreviewAsync } from 'actions/jobs-actions';
+import { getTaskPreviewAsync } from 'actions/tasks-actions';
+import { getProjectsPreviewAsync } from 'actions/projects-actions';
+import { getCloudStoragePreviewAsync } from 'actions/cloud-storage-actions';
+import { getModelPreviewAsync } from 'actions/models-actions';
+
+export type PreviewEntity = Job | Task | Project | CloudStorage | MLModel;
+export type PreviewType = 'job' | 'task' | 'project' | 'cloudStorage' | 'model';
+
+export function getPreviewType(entity: PreviewEntity): PreviewType {
+    if (entity instanceof Job) return 'job';
+    if (entity instanceof Task) return 'task';
+    if (entity instanceof Project) return 'project';
+    if (entity instanceof CloudStorage) return 'cloudStorage';
+    if (entity instanceof MLModel) return 'model';
+    throw new Error('Unknown entity type');
+}
+
+export function getRequestId(entity: PreviewEntity): string {
+    const type = getPreviewType(entity);
+    return `${type}-${(entity as any).id}`;
+}
+
+interface PreviewRequest {
+    id: string;
+    entity: PreviewEntity;
+    dispatch: Dispatch;
+}
+
+const PREVIEW_REQUEST_DELAY = 100;
+
+class PreviewQueue {
+    private queue: PreviewRequest[] = [];
+    private isProcessing: boolean = false;
+    private currentRequest: PreviewRequest | null = null;
+
+    public addRequest(request: PreviewRequest): void {
+        const isDuplicate = this.queue.some((req) => req.id === request.id) ||
```

Comment:
If we receive request for another entity type, it usually means that user changed the page. 
Maybe in this case it would be better to push new entity in the beginning of the queue.

Reply:
Currently, if the preview component is unmounted before its request is processed, the request is removed from the queue entirely. This means that previews for new entities start fetching immediately.

Do you think we should avoid aborting previous requests and instead move them to the end of the queue?

Reply:
Good